ltiauthenticator/lti11/auth.py

```diff
@@ -78,16 +97,16 @@ async def authenticate(  # noqa: C901
         self.log.debug("Launch url is: %s" % launch_url)
 
         if validator.validate_launch_request(launch_url, handler.request.headers, args):
-            # get the lms vendor to implement optional logic for said vendor
-            canvas_id = handler.get_body_argument("custom_canvas_user_id", default=None)
-
-            if canvas_id is not None:
-                user_id = handler.get_body_argument("custom_canvas_user_id")
+            username = ""
```

Comment:
```suggestion
```

Not actually needed, since we never want to return "" as a username. If `username` isn't defined in all the arms of the `if` statement below, we'll get a clean reference error than dig through to find out why username is `""`

Reply:
Of course! Thanks for pointing that out.